cmd/clairctl/clair/analyse.go

```diff
@@ -0,0 +1,39 @@
+package clair
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/coreos/clair/api/v1"
+)
+
+//Analyse get Analysis os specified layer
+func Analyse(id string) (v1.LayerEnvelope, error) {
+
+	lURI := fmt.Sprintf("%v/layers/%v?vulnerabilities", uri, id)
+	// lURI := fmt.Sprintf("%v/layers/%v/vulnerabilities?minimumPriority=%v", uri, id, priority)
+	response, err := http.Get(lURI)
+	if err != nil {
+		return v1.LayerEnvelope{}, fmt.Errorf("analysing layer %v: %v", id, err)
+	}
+	defer response.Body.Close()
+
+	body, err := ioutil.ReadAll(response.Body)
```

Comment:
You don't need to read the entire body. You can simply use `json.NewDecoder(response.Body).Decode(&analysis)`. You should still test the error code though.